minimint/src/consensus/mod.rs

```diff
@@ -342,6 +402,13 @@ where
             )
             .collect();
 
+        if let Some(epoch) = self.db.get_value(&LastEpochKey).unwrap() {
+            let last_epoch = self.db.get_value(&epoch).unwrap().unwrap();
+            let sig = self.cfg.hbbft_sks.0.sign(last_epoch.hash);
```

Comment:
Reusing the same key is not necessarily safe. At the least we should use tagged hashing and check that HBBFT does the same so signatures can't be re-used between protocols. It might be easier to just generate yet another key set or use naive, concatenated schnorr-based threshold signatures.

Reply:
I'm not sure why it isn't safe if we are signing a cryptographic hash that contains the same information that was in the HBBFT epoch.  Are you saying it's possible an attacker could construct an epoch that has an identical message to be signed?